pkg/mq/msgstream/msgstream.go

```diff
@@ -52,6 +57,238 @@ type MsgPack struct {
 	EndPositions   []*MsgPosition
 }
 
+// ConsumeMsgPack represents a batch of msg in consumer
+type ConsumeMsgPack struct {
+	BeginTs        Timestamp
+	EndTs          Timestamp
+	Msgs           []PackMsg
+	StartPositions []*MsgPosition
+	EndPositions   []*MsgPosition
+}
+
+// PackMsg used for ConumserMsgPack
+// support fetch some properties metric
+type PackMsg interface {
```

Comment:
Maybe rename to ConsumedMsg, ConsumedMsgPack.

PackMsg is too confused with MsgPack.